web3.js/test/connection.test.ts

```diff
@@ -1513,6 +1513,98 @@ describe('Connection', function () {
     expect(result).to.be.empty;
   });
 
+  it('get block production', async () => {
+    const blockProductionCommitment: Commitment = 'confirmed';
+
+    const blockProductionConfig = {
+      range: {
+        firstSlot: 0,
+        lastSlot: 10,
```

Comment:
So close on the live validator test! This can _still_ fail if, for instance, this was the first test to run.

I was able to break this by changing `it` to `it.only` then running `yarn test:live-with-test-validator`. Since it was the first test to run, block 10 didn't exist yet!

```
Error: failed to get block production information: lastSlot, 10, is too large; max 0
```

Consider an approach, in this test, where you:

1. Use `getBlockHeight()` and `getFirstAvailableBlock()` to make your choice of `firstSlot` and `lastSlot`
2. Mock out the `getBlockHeight` and `getFirstAvailableBlock` response in the non-live case
3. Make sure that `firstSlot` isn't too far away from `lastSlot` (read the Rust code to figure out what the maximum distance is; I forget!